python-capdl-tool/capdl/Object.py

```diff
@@ -370,12 +370,23 @@ def __repr__(self):
     def set_affinity(self, affinity):
         self.affinity = affinity
 
-    def set_fault_ep_slot(self, fault_ep_slot=0, fault_ep=None, badge=0):
+    def set_fault_ep_slot(self, fault_ep_slot=0, fault_ep=None, badge=0, rights=ObjectRights.seL4_AllRights):
         if fault_ep_slot != 0:
             self.fault_ep_slot = fault_ep_slot
         if fault_ep:
+            fields = []
             if badge != 0:
-                fault_ep += " (badge: %d)" % badge
+                fields += ['badge: %d' % badge]
+            rights_list = [sym for sym, flag in
```

Comment:
could avoid some redundancy:
```
rights_list = [
    sym for sym, right in [
        ('R', ObjectRights.seL4_CanRead),
        ('W', ObjectRights.seL4_CanWrite),
        ('G', ObjectRights.seL4_CanGrant),
        ('P', ObjectRights.seL4_CanGrantReply)
    ] if right in rights
]
```

and as a single (python magic) statement this would work, but it tends to a bit unreadable
```
fields = ','.join([
    s for s in [ 
        ('badge: %d' % badge) if badge != 0 else '',
        ''.join([ sym for sym, right in [
            ('R', ObjectRights.seL4_CanRead),
            ('W', ObjectRights.seL4_CanWrite),
            ('G', ObjectRights.seL4_CanGrant),
            ('P', ObjectRights.seL4_CanGrantReply)
        ] if right in rights ])
    ] if s ])
```

Reply:
Thanks, that's much better!

I have to admit that as someone without that much python experience, the second option is a bit too far on the unreadable magic side for my preference.

Reply:
Yes, the second option is terrible, don't use that :-)

Reply:
(I do like the first)